File: app/src/main/java/com/example/camera/base/BaseActivity.java
package com.example.camera.base;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import butterknife.ButterKnife;

/**
 * @author LiMing
 * @Demo class BaseActivity
 * @Description TODO
 * @date 2019-09-25 14:03
 */
public abstract class BaseActivity extends AppCompatActivity {

    protected Activity context;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        context = this;
        onCreateView();
        ButterKnife.bind(this);
        initView(savedInstanceState);
    }

    protected abstract void onCreateView();

    protected abstract void initView(@Nullable Bundle savedInstanceState);
}

File: KotlinCameraX/src/main/java/com/example/camerax/base/BaseActivity.kt
package com.example.camerax.base

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * @Demo class BaseActivity
 * @Description TODO
 * @author libo
 * @date 2019-11-19 11:26
 */
abstract class BaseActivity :AppCompatActivity() {

    lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        context = this;
        onCreateView()
        initView(savedInstanceState)
    }

    abstract fun onCreateView()

    abstract fun initView(savedInstanceState: Bundle?)

}